docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,328 @@
+```````````````````````````````````````
+
+.. currentmodule:: chainer
+
+In this tutorial section, we will learn how to train a deep neural network to classify images of hand-written digits in the popular MNIST dataset. This dataset contains 50,000 training examples and 10,000 test examples. Each example is a set of a 28 x 28 greyscale image and a corresponding class label. Since the digits from 0 to 9 are used, there are 10 classes for the labels.
+
+Chainer provides a feature called :class:`~chainer.training.Trainer` that can simplify the training procedure of your model. However, it is also good to know how the training works in Chainer before starting to use the useful :class:`~chainer.training.Trainer` class that hides the actual processes. For advanced users, it sometimes needs to write their own custom training loop.
+
+The complete training procedure consists of the following steps:
+
+1. Prepare a dataset.
+2. Create an iterator for the dataset.
+3. Write a training loop that performs the following operations in each iteration:
+    1. Retrieve a set of examples (mini-batch) from the training dataset.
+    2. Feed the mini-batch to your network.
+    3. Run a forward pass of the network and compute the loss.
+    4. Just call the :meth:`~chainer.Variable.backward` method from the loss :class:`~chainer.Variable` to compute the gradients for all trainable parameters.
+    5. Run the optimizer to update those parameters.
+    6. (Optional): Check the network performance on the validation/test set.
+
+1. Prepare the dataset
+''''''''''''''''''''''
+
+Chainer contains some built-in functions to use some popular datasets like MNIST, CIFAR10/100, etc. Those can automatically download the data from servers and provide dataset objects which are easy to use.
+
+The below code shows how to retrieve the MNIST dataset from the server and save an image from its training split to make sure the images are correctly obtained.
+
+.. testcode::
+
+    import matplotlib.pyplot as plt
+    from chainer.datasets import mnist
+
+    # Download the MNIST data if you haven't downloaded it yet
+    train, test = mnist.get_mnist(withlabel=True, ndim=1)
+
+    # Display an example from the MNIST dataset.
+    # `x` contains the input image array and `t` contains that target class
+    # label as an integer.
+    x, t = train[0]
+    plt.imshow(x.reshape(28, 28), cmap='gray')
+    plt.savefig('5.png')
+    print('label:', t)
+
+.. testoutput::
+
+    label: 5
+
+The saved image ``5.png`` will look like:
+
+.. image:: ../../image/train_loop/5.png
+
+
+2. Create the dataset iterators
+'''''''''''''''''''''''''''''''
+
+Although this is an optional step, we'd like to introduce the :class:`~chainer.dataset.Iterator` class that enables to retrieve a set of data and labels from the given dataset and give a mini-batch easily. There are some subclasses that can perform the same thing in different ways, e.g., using multi-processing to parallelize the data loading part, etc.
```

Comment:
Although this is an optional step, we'd like to introduce the :class:`~chainer.dataset.Iterator` class that retrieves a set of data and labels from the given dataset to easily make a mini-batch. There are some subclasses that can perform the same thing in different ways, e.g., using multi-processing to parallelize the data loading part, etc.